### ISSUE TITLE
False positive: db.Queryx

### ISSUE THREAD

**Issue by pellared** [ryanrolds/sqlclosecheck#35]
Version: `v0.5.1`

Repro steps:

```go
package bug_test

import (
	"log"

	"github.com/jmoiron/sqlx"
)

func Example() {
	db, err := sqlx.Open("pgx", "postgres://localhost/db")
	if err != nil {
		log.Print(err)
		return
	}
	defer db.Close()

	rows, err := db.Queryx("SELECT * FROM users")
	if err != nil {
		log.Print(err)
		return
	}
	defer rows.Close()
}
```

Result (from `golangci-lint`):

```shell
example_test.go:17:24: Rows/Stmt/NamedStmt was not closed (sqlclosecheck)
        rows, err := db.Queryx("SELECT * FROM users")
```

**Comment by shubhamzanwar** [comment]
➕1

we're facing the same issue in our workflows

**Comment by ryanrolds** [comment]
Thank you for the report. I will take a look at this over the next few days.

**Comment by shubhamzanwar** [comment]
How's it looking with this one? 🙏🏽

**Comment by Bodyancheq** [comment]
+1 
in my project following code worked as a solution:
`
rows, err := db.Queryx("SELECT * FROM users")
if err != nil {
  log.Print(err)
  return
}
defer func() {
  _ = rows.Close()
}()
`

also worth mentioning: when I used sql instead of sqlx, there was no error

**Comment by ivanovaleksey** [comment]
Hi @ryanrolds ! Is there any progress on this issue?

**Comment by jordicaballero** [comment]
Hi, I'm still having this issue under different `golangci-lint` versions. Using `jmoiron/sqlx` library:

```go
	rows, err := sqlx.NamedQueryContext(ctx, r.connection, query, dbOrg) //nolint:sqlclosecheck 😢 
       // also happens using -> rows, err := r.connection.QueryxContext(ctx, query, dbOrgRel)
	if err != nil {
          ....
	}
	defer rows.Close()
	if rows.Err() != nil {
		...
	}
	for rows.Next() {
	... 
	}
```

I tried to debug locally, but I cannot reproduce. Our repository is big. I tested isolating this code into `testdata` folder, but it works fine.

I should put the //nocheck by the time.
